dspace-server-webapp/src/test/java/org/dspace/app/rest/StatisticsRestRepositoryIT.java

```diff
@@ -1206,6 +1225,181 @@ public void usageReportsSearch_Bitstream_Visited() throws Exception {
+            Pair.of("type", COMMUNITY),
+            Pair.of("ip", "5b02:f3ed:635f:98b1:d2c5:f292:90d9:3982"),
+            Pair.of("dns", "dns_4"),
+            Pair.of("time", getTimeNDaysAgo(89))
+        ));
+
+        // bitstream document which should be anonymized
+        addSolrDocument(asList(
+            Pair.of("id", "bitstream_view_old"),
+            Pair.of("type", BITSTREAM),
+            Pair.of("ip", "75.133.248.54"),
+            Pair.of("dns", "dns_1"),
+            Pair.of("time", getTimeNDaysAgo(90))
+        ));
+
+        // item document which should be anonymized
+        addSolrDocument(asList(
+            Pair.of("id", "item_view_old"),
+            Pair.of("type", ITEM),
+            Pair.of("ip", "195.11.13.244"),
+            Pair.of("dns", "dns_2"),
+            Pair.of("time", getTimeNDaysAgo(130))
+        ));
+
+        // collection document which should be anonymized
+        addSolrDocument(asList(
+            Pair.of("id", "collection_view_old"),
+            Pair.of("type", COLLECTION),
+            Pair.of("ip", "16f4:0586:3148:3a8a:f307:e13e:2614:21a2"),
+            Pair.of("dns", "dns_3"),
+            Pair.of("time", getTimeNDaysAgo(200))
+        ));
+
+        // community document which should be anonymized
+        addSolrDocument(asList(
+            Pair.of("id", "community_view_old"),
+            Pair.of("type", COMMUNITY),
+            Pair.of("ip", "5b02:f3ed:635f:98b1:d2c5:f292:90d9:3982"),
+            Pair.of("dns", "dns_4"),
+            Pair.of("time", getTimeNDaysAgo(500))
+        ));
+
+        solrStatisticsCore.getSolr().commit();
+
+        runDSpaceScript("anonymize-statistics");
+
+        assertEquals(
+            "75.133.248.54",
+            getSolrDocumentById("bitstream_view_recent").getFieldValue("ip")
+        );
+        assertEquals(
+            "dns_1",
+            getSolrDocumentById("bitstream_view_recent").getFieldValue("dns")
+        );
+
+        assertEquals(
+            "195.11.13.244",
+            getSolrDocumentById("item_view_recent").getFieldValue("ip")
+        );
+        assertEquals(
+            "dns_2",
+            getSolrDocumentById("item_view_recent").getFieldValue("dns")
+        );
+
+        assertEquals(
+            "16f4:0586:3148:3a8a:f307:e13e:2614:21a2",
+            getSolrDocumentById("collection_view_recent").getFieldValue("ip")
+        );
+        assertEquals(
+            "dns_3",
+            getSolrDocumentById("collection_view_recent").getFieldValue("dns")
+        );
+
+        assertEquals(
+            "5b02:f3ed:635f:98b1:d2c5:f292:90d9:3982",
+            getSolrDocumentById("community_view_recent").getFieldValue("ip")
+        );
+        assertEquals(
+            "dns_4",
+            getSolrDocumentById("community_view_recent").getFieldValue("dns")
+        );
+
+        assertEquals(
+            "75.133.248.255",
+            getSolrDocumentById("bitstream_view_old").getFieldValue("ip")
```

Comment:
This is a good test method overall...but it's *very* difficult to tell here that the IP address _is actually_ anonymized.  Could you add a comment here to note that the IP was anonymized to end in 255?  or you could use a variable for the first 3 parts of the IP address to show only the *ending* changes.  

It literally took me 5 mins to realize that this IP *is* anonymized.  I originally started writing a comment here to ask why this isn't anonymized at all. :) 

So, I think these tests are good enough to prove things are working. They just could use a few inline comments (or use of variables) to better describe how the anonymization is occurring, so that they are easier for others to maintain & understand.